### PR TITLE
Add tests for atoms heading and atomss heading icom

### LINKED ISSUE #14: Backfill tests for Atoms/Heading and Atoms/HeadingIcon
Using other component tests as a guide, write automated tests to cover important scenarios in component. Try to avoid granular assertions of the markup. Stay as high-level as possible to avoid test fragility.

### PR DESCRIPTION
**Pull request by SaidShah**
- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)
- [x] New component and/ or properties? Make sure to update storybook

Closes #14

### CODE REVIEW

**Review by jstevenson — APPROVED** [review]

File: src/atoms/typography/heading-icon.test.tsx [hunk 1]
```diff
@@ -1,5 +1,20 @@
 import React from "react";
+import faker from "faker";
+import { HeadingIcon } from "./heading-icon";
+import { render } from "@testing-library/react";
+import { Icons } from "../constants/icons";
 
 describe("HeadingIcon", () => {
-    test.skip("TODO - https://github.com/AndcultureCode/AndcultureCode.JavaScript.React.Components/issues/14", () => {});
+    test("when default props, renders Heading with Icon", () => {
```
Comment (brandongregoryscott) [inline src/atoms/typography/heading-icon.test.tsx:8]: Does this actually test that an icon was rendered? I'd update the test name or add an additional test to cover that an icon is also rendered.

File: src/atoms/typography/heading.test.tsx [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
+        const expected = faker.random.words();
+
+        // Act
+        const { getByText } = render(<Heading>{expected}</Heading>);
+
+        // Assert
+        expect(getByText(expected)).not.toBeNil();
+    });
+
+    test("when given cssClassName prop, renders with given class name", () => {
```
Comment (brandongregoryscott) [inline src/atoms/typography/heading.test.tsx:18]: Recommend naming this `when cssClassName prop has value, renders with class name`

**Review by brandongregoryscott — CHANGES_REQUESTED** [review]
Left some minor feedback to adjust @SaidShah

File: src/atoms/typography/heading.test.tsx (hunk 2 above)
Comment (SaidShah) [reply src/atoms/typography/heading.test.tsx]: I updated the tests if you want to take a second look. 👍

**Review by brandongregoryscott — APPROVED** [review]

**Pull request merged**